### PR TITLE
Improve computing of available space for responsive layouts

### LINKED ISSUE #7993: Improve computing of available space for responsive layouts

### PR DESCRIPTION
**Pull request by mattpap**
fixes #7993

### CODE REVIEW

File: bokehjs/src/lib/models/layouts/layout_dom.ts [hunk 1]
```diff
@@ -92,8 +92,10 @@ export abstract class LayoutDOMView extends DOMView {
       const {left, right, top, bottom} = padding(measuring)
       const {width, height} = measuring.getBoundingClientRect()
 
-      if (width != 0 || height != 0)
-        return [width - left - right, height - top - bottom]
+      const inner_width = width - left - right
+      const inner_height = height - top - bottom
+      if (inner_width > 0 && inner_height > 0)
```
Comment (mattpap) [inline bokehjs/src/lib/models/layouts/layout_dom.ts]: Changing `||` to `&&` will likely break `scale_width` and `scale_height`.

**Comment by bryevdv** [comment]
@mattpap What else do you plan to do for this PR?

**Comment by mattpap** [comment]
I don't plan any new developments, except not breaking existing code (which this PR currently does). I definitively want to finish this today.

**Comment by mattpap** [comment]
@bryevdv, this is "ready". I doubt I can improve things further without a proper redesign of the layout. 

Currently the biggest problem with simple layouts with `*_both` modes, is having `absolute` positioning required on roots having those sizing modes. This makes templates fragile, because you have to make sure you don't trip over `absolute` positioning, which completely disregards the flow of content on a page. This limitation needs to be lifted, but this will require some work.

**Pull request merged**